### PR TITLE
Implement the estimation of maximum number of inputs for Shelley and show big wallet migration

### PR DESCRIPTION
**Pull request by paweljakubas**
# Issue Number

#1751 

# Overview

- [x] I have refactored computeTxSize to use later
- [x] I have implemented `_estimateMaximumNumberOfInputs`
- [x] I have unpended integration test showing the case - big wallet migration
- [x] I have added property tests

# Comments

### CODE REVIEW

**Comment by paweljakubas** [comment]
bors try

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs [hunk 1]
```diff
@@ -246,10 +242,9 @@ spec = do
             Empty >>= flip verify
             [ expectField
                 #distribution
-                ((`shouldBe` (Just 400)) . Map.lookup 10_000_000_000)
+                ((`shouldBe` (Just 129)) . Map.lookup 100_000_000_000)
```
Comment (piotr-iohk) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs]: This does not appear to be right... :thinking: 
The fixture wallet has 400 utxo * 100_000_000_000 + dust (100 utxo * 1), so I'd expect target wallet ditribution to be 400, not 129...

https://github.com/input-output-hk/cardano-wallet/blob/master/lib/shelley/test/data/cardano-node-shelley/genesis.yaml#L1217

Comment (paweljakubas) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs:245]: probably addresses were reused and hence we have decreased number of utxos, BUT balance is ok. if so the test should be corrected in such a way that we are sure 400 distinct addresses are used in the migration. That's my guess

Comment (paweljakubas) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs]: I do not confirm original balance 400 * ...
I see : 
```
################## originalBalance:12900000000000
################## expectedFee:3574500
################## expectedBalance:12899996425500
```

Comment (piotr-iohk) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs]: Yes, but wallet from the genesis file has the following: 400 * 100_000_000_000 and 100 * 1... so  the original balance should be 40000000000500... :thinking:

**Comment by paweljakubas** [comment]
bors try

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs]: I think it should be subject of another BUG ...

**Review by KtorZ — APPROVED** [review]

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs (hunk 1 above)
Comment (KtorZ) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs]: The problem comes from:

a) Funds are defined as a key-value "map" in the genesis file. 
b) This is therefore parsed as a map by ledger's functions & al. 
c) These identify _sequential_ addresses, that need to be discovered in order! However, the map "destroys" any sort of order we could have on keys... and there's no guarantee that we process addresses in order. Actually, we don't, and are only able to discover 129 of those addresses.

Comment (KtorZ) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs]: Was harder than I thought, but I got it:

```
(Status {statusCode = 200, statusMessage = "OK"},Right (ApiUtxoStatistics {total = Quantity {getQuantity = 40000000000100}, scale = ApiT {getApiT = Log10}, distribution = fromList [(10,100),(100,0),(1000,0),(10000,0),(100000,0),(1000000,0),(10000000,0),(100000000,0),(1000000000,0),(10000000000,0),(100000000000,400),(1000000000000,0),(10000000000000,0),(100000000000000,0),(1000000000000000,0),(10000000000000000,0),(45000000000000000,0)]}))
```

**Comment by KtorZ** [comment]
bors retry

**Comment by KtorZ** [comment]
bors retry

**Comment by paweljakubas** [comment]
bors retry

**Comment by paweljakubas** [comment]
once again : 
```
Progress 6/7: cardano-wallet-shelley-2020.6.5
Timeout after 45 minutes.
```

**Comment by KtorZ** [comment]
bors retry

**Comment by KtorZ** [comment]
bors retry

**Pull request merged**